cli/pcluster/config_sanity.py

```diff
@@ -530,6 +530,56 @@ def validate(self, resource_type, resource_value):  # noqa: C901 FIXME
         # FSX FS Id check
         elif resource_type in ["fsx_fs_id", "FSx_storage_capacity", "FSx_imported_file_chunk_size", "FSx_export_path"]:
             self.__validate_fsx_parameters(resource_type, resource_value)
+        elif resource_type == "EFA":
+            valid_instances = get_supported_efa_instances(self.region)
+            if "ComputeInstanceType" in resource_value:
+                if resource_value.get("ComputeInstanceType") not in valid_instances:
+                    self.__fail(resource_type, "Compute Instance needs to be one of %s" % valid_instances)
+            else:
+                self.__fail(resource_type, "Compute Instance needs to be one of %s" % valid_instances)
+            if resource_value.get("PlacementGroup", "NONE") == "NONE":
+                self.__fail(resource_type, "Placement group is required, set placement_group.")
+            if "VPCSecurityGroupId" in resource_value:
+                sg_id = resource_value.get("VPCSecurityGroupId")
+                try:
+                    ec2 = boto3.client(
+                        "ec2",
+                        region_name=self.region,
+                        aws_access_key_id=self.aws_access_key_id,
+                        aws_secret_access_key=self.aws_secret_access_key,
+                    )
+                    sg = ec2.describe_security_groups(GroupIds=[sg_id]).get("SecurityGroups")[0]
+                    in_rules = sg.get("IpPermissions")
+                    out_rules = sg.get("IpPermissionsEgress")
+
+                    allowed_in = False
+                    allowed_out = False
+                    for rule in in_rules:
+                        if (
+                            rule.get("IpProtocol") == "-1"
+                            and len(rule.get("UserIdGroupPairs")) > 0
+                            and rule.get("UserIdGroupPairs")[0].get("GroupId") == sg_id
+                        ):
+                            allowed_in = True
+                            break
+                    for rule in out_rules:
+                        if (
+                            rule.get("IpProtocol") == "-1"
+                            and len(rule.get("UserIdGroupPairs")) > 0
+                            and rule.get("UserIdGroupPairs")[0].get("GroupId") == sg_id
+                        ):
+                            allowed_out = True
+                            break
+                    if not (allowed_in and allowed_out):
+                        self.__fail(
+                            resource_type,
+                            "VPC Security Group %s must allow all traffic in and out from itself. "
+                            "See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start.html#efa-start-security"
+                            % sg_id,
+                        )
+                except ClientError as e:
+                    self.__fail(resource_type, e.response.get("Error").get("Message"))
```

Comment:
Since we don't have any stacktrace for this failures let's maybe leave an additional message such as failed while validating EFA resources with exception ....

Reply:
In the `fail()` function we print out the `resource_type`. For example, a misconfigured security group results in:

```bash
$ pcluster create efa
Beginning cluster creation for cluster: efa
Config sanity error on resource EFA: VPC Security Group sg-00e715cfdebcb97bb must allow all traffic in and out from itself. See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start.html#efa-start-security
```